### DISCUSSION TITLE
What are the required privileges for argocd-manager-role in a remote cluster?

### DISCUSSION THREAD

**Discussion by gaeljw** [argoproj/argo-cd#15821] [Q&A]
When I added a remote cluster to ArgoCD using `argocd cluster add`, I didn't notice but the created `ClusterRole` `argocd-manager-role` has been created with too many privileges (not sure why, is it because my local kubeconfig had too many privileges?).

Anyway, I'd like to review it and only give it the mandatory privileges. Where can I find such a list? Should it be similar to the `argocd-server` role on a local cluster?

**Comment by dtwilliamsWork** [comment]
from https://argo-cd.readthedocs.io/en/stable/getting_started/

The above command installs a ServiceAccount (argocd-manager), into the kube-system namespace of that kubectl context, and binds the service account to an admin-level ClusterRole. Argo CD uses this service account token to perform its management tasks (i.e. deploy/monitoring).

The rules of the argocd-manager-role role can be modified such that it only has create, update, patch, delete privileges to a limited set of namespaces, groups, kinds. However get, list, watch privileges are required at the cluster-scope for Argo CD to function.

**Reply by luwangVMW** [reply]
@dtwilliamsWork  Is there a method to create argocd-manager-role with expected privileges when using argocd cli to manage the cluster? we do not want to modify it after the role is created/updated by argocd, due to some of users do not have such permission to do it.

**Reply by luwangVMW** [reply]
Check the code,  it seems not provide such function. 
`func upsertRole(clientset kubernetes.Interface, name string, namespace string, rules []rbacv1.PolicyRule) error {
	role := rbacv1.Role{
		TypeMeta: metav1.TypeMeta{
			APIVersion: "rbac.authorization.k8s.io/v1",
			Kind:       "Role",
		},
		ObjectMeta: metav1.ObjectMeta{
			Name: name,
		},
		Rules: rules,
	}
	return upsert("Role", fmt.Sprintf("%s/%s", namespace, name), func() (any, error) {
		return clientset.RbacV1().Roles(namespace).Create(context.Background(), &role, metav1.CreateOptions{})
	}, func() (any, error) {
		return clientset.RbacV1().Roles(namespace).Update(context.Background(), &role, metav1.UpdateOptions{})
	})
}`

May I know what is the minimum permission for the argocd if it manages cluster with namespace scope?
